### ISSUE TITLE
Heredoc after source causes sourced file to fail to parse (SC1094)

### ISSUE THREAD

**Issue by ruuda** [koalaman/shellcheck#2803]
#### For bugs
- Rule Id: SC1094: 
- My shellcheck version: 0.9.0
- [x] The rule's wiki page does not already cover this
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit: I was unable to because it looks like on shellcheck.net, I can only enter a single file.

#### Here's a snippet or screenshot that shows the problem:

a.sh:
```sh
#!/usr/bin/bash
source ./b.sh <<EOF
hi
EOF
```

b.sh:
```sh
#!/usr/bin/bash
cat
```

Bash can definitely parse b.sh, and Shellcheck does not report any errors on it when I run it on only b.sh.
```
$ ./a.sh
hi
```

#### Here's what shellcheck currently says:

```console
$ shellcheck b.sh

$ shellcheck a.sh b.sh

In a.sh line 2:
source ./b.sh <<EOF
       ^----^ SC1094 (warning): Parsing of sourced file failed. Ignoring it.
```

#### Here's what I wanted or expected to see:

Shellcheck should not fail to parse b.sh because of the heredoc. When I replace the heredoc with e.g. `< /dev/null`, the parse error goes away, so it looks like it’s really the heredoc causing a problem here.

**Comment by koalaman** [comment]
This was caused by ShellCheck unintentionally propagating the current stack of pending here documents, which caused the sourced script to look for them in the sourced file. It's now fixed by clearing and restoring the pending here docs before sourcing.

Nice find, thanks!

**Comment by ruuda** [comment]
Thanks for the fix!
